pkg/controlplane/controller/defaultservicecidr/default_servicecidr_controller.go

```diff
@@ -0,0 +1,219 @@
+		interval: 10 * time.Second, // same as DefaultEndpointReconcilerInterval
+	}
+
+	// obtain configuration from flags
+	if netutils.IsIPv4CIDR(&primaryRange) {
+		c.ipv4CIDR = primaryRange.String()
+	} else if netutils.IsIPv4CIDR(&secondaryRange) {
+		c.ipv4CIDR = secondaryRange.String()
+	}
+	if netutils.IsIPv6CIDR(&primaryRange) {
+		c.ipv6CIDR = primaryRange.String()
+	} else if netutils.IsIPv6CIDR(&secondaryRange) {
+		c.ipv6CIDR = secondaryRange.String()
+	}
+	// we construct our own informer because we need such a small subset of the information available.
+	// The kubernetes.default service
+	c.serviceCIDRInformer = networkingv1alpha1informers.NewFilteredServiceCIDRInformer(client, 12*time.Hour,
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+		func(options *metav1.ListOptions) {
+			options.FieldSelector = fields.OneTermEqualSelector("metadata.name", DefaultServiceCIDRName).String()
+		})
+
+	c.serviceCIDRLister = networkingv1alpha1listers.NewServiceCIDRLister(c.serviceCIDRInformer.GetIndexer())
+	c.serviceCIDRsSynced = c.serviceCIDRInformer.HasSynced
+
+	c.eventBroadcaster = broadcaster
+	c.eventRecorder = recorder
+
+	c.readyCh = make(chan struct{})
+
+	return c
+}
+
+// Controller manages selector-based service ipAddress.
+type Controller struct {
+	ipv4CIDR string
+	ipv6CIDR string
+
+	client           clientset.Interface
+	eventBroadcaster record.EventBroadcaster
+	eventRecorder    record.EventRecorder
+
+	serviceCIDRInformer cache.SharedIndexInformer
+	serviceCIDRLister   networkingv1alpha1listers.ServiceCIDRLister
+	serviceCIDRsSynced  cache.InformerSynced
+
+	readyCh chan struct{} // channel to block until the default ServiceCIDR exists
+
+	interval time.Duration
+}
+
+// Start will not return until the default ServiceCIDR exists or stopCh is closed.
+func (c *Controller) Start(stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+
+	c.eventBroadcaster.StartStructuredLogging(0)
+	c.eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: c.client.CoreV1().Events("")})
+	defer c.eventBroadcaster.Shutdown()
+
+	klog.Infof("Starting %s", controllerName)
+	defer klog.Infof("Shutting down %s", controllerName)
+
+	go c.serviceCIDRInformer.Run(stopCh)
+	if !cache.WaitForNamedCacheSync(controllerName, stopCh, c.serviceCIDRsSynced) {
+		return
+	}
+
+	go wait.Until(c.sync, c.interval, stopCh)
+
+	select {
+	case <-stopCh:
+	case <-c.readyCh:
+	}
+
+}
+
+func (c *Controller) sync() {
+	// check if the default ServiceCIDR already exist
+	serviceCIDR, err := c.serviceCIDRLister.Get(DefaultServiceCIDRName)
+	// if exists
+	if err == nil {
+		c.syncStatus(serviceCIDR)
+		return
+	}
+
+	// unknown error
+	if !apierrors.IsNotFound(err) {
+		klog.Infof("error trying to obtain the default ServiceCIDR: %v", err)
+		return
+	}
+
+	// default ServiceCIDR does not exist
+	klog.Infof("Creating default ServiceCIDR, ipv4: %s ipv6: %s", c.ipv4CIDR, c.ipv6CIDR)
+	serviceCIDR = &networkingapiv1alpha1.ServiceCIDR{
```

Comment:
yes , is implicitly mentioned, the behavior of ServiceCIDR is the same as for the kubernetes.default Service, if it exists no apiserver  touches it, in this cse it will happen the same, the apiserver will not create the new one if there is still one created, the unique name provides us consistency